### PR TITLE
Expand readme (inspired by server-repo)

### PR DESCRIPTION
**Pull request by tim-vk**
A set-up for a more complete README. I did notice some oddities while writing it though:

- Current `latest` version is 1.0.1 however there is also a version 1.1.0 version available which seems odd. Is this a mistake? (I do think 1.0.1 is newer, but the number is off. Would assume it to be 1.1.1 or 1.2.0).
- The bit about `testing`: We can switch between a stable & unstable version. But no where can I actually find an actual implementation of something that does something with this setting.  The jellyfin-web client seems to use it's own chromecast plugin. So is this repo even used a.t.m.? (if so: can you point me to the right direction so I can figure out stuff further from there?).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,37 @@
-# Jellyfin Chromecast
+<h1 align="center">Jellyfin Chromecast Client</h1>
```
Comment (thornbill) [inline README.md:1]: Jellyfin Chromecast Web Receiver or Jellyfin Chromecast Receiver App might be better names?

File: README.md [hunk 2]
```diff
@@ -1,3 +1,37 @@
...
+<a href="https://github.com/jellyfin/jellyfin/releases">
+<img alt="Current Release" src="https://img.shields.io/github/release/jellyfin/jellyfin-chromecast.svg"/>
+</a>
+</p>
+
+Jellyfin chromecast is a receiver used when casting to a google cast capable device. It is automatically used when casting from the jellyfin android client or jellyfin web client.
+
+### What does it do?
+
+As soon as you push the "cast" button on your client this application will start on you cast-capable device & handle it from there.
```
Comment (thornbill) [inline README.md:20]: It would probably be good to provide this link for reference to the overall casting process/architecture: https://developers.google.com/cast/docs/overview

File: README.md [hunk 3]
```diff
@@ -1,3 +1,37 @@
...
+
+### What doesn't it do?
+
+Anything related to your non-cast device (e.g. your phone, browser, other device). Any issues/features related to that: check the respective repository.
```
Comment (thornbill) [inline README.md:24]: It would be good to specifically point out that this is not related to clients supporting casting either. i.e. Don't open issues here about the iOS app not supporting casting.

File: README.md [hunk 4]
```diff
@@ -1,3 +1,37 @@
...
+<br/>
+<a href="https://github.com/jellyfin/jellyfin">
+<img alt="GPL 2.0 License" src="https://img.shields.io/github/license/jellyfin/jellyfin-chromecast.svg"/>
+</a>
+<a href="https://github.com/jellyfin/jellyfin/releases">
+<img alt="Current Release" src="https://img.shields.io/github/release/jellyfin/jellyfin-chromecast.svg"/>
+</a>
+</p>
+
+Jellyfin chromecast is a receiver used when casting to a google cast capable device. It is automatically used when casting from the jellyfin android client or jellyfin web client.
```
Comment (thornbill) [inline README.md:16]:
```suggestion
Jellyfin Chromecast is a receiver app used when casting to a Google Cast capable device. It is used when casting from the Jellyfin Android client or Jellyfin web client.
```

It's probably also worth noting that the default instance of the app is hosted by Jellyfin.

File: README.md [hunk 5]
```diff
@@ -1,3 +1,37 @@
...
+### What does it do?
+
+As soon as you push the "cast" button on your client this application will start on you cast-capable device & handle it from there.
+
+### What doesn't it do?
+
+Anything related to your non-cast device (e.g. your phone, browser, other device). Any issues/features related to that: check the respective repository.
+
+### Something not working right?
+First check if the issue is actually google-cast related. So answer the question:
```
Comment (thornbill) [inline README.md:27]:
```suggestion
First check if the issue is actually Google Cast related. So answer the question:
```

File: README.md [hunk 6]
```diff
@@ -1,3 +1,37 @@
...
+
+`"Can I reproduce the issue on any other device than a google chromecast?"`
```
Comment (thornbill) [inline README.md:29]:
```suggestion
> "Can I reproduce the issue on any other device than a Google Chromecast?"
```

File: README.md [hunk 7]
```diff
@@ -1,3 +1,37 @@
...
+
+If yes: Then the issue probably lies somewhere else. If no: Open an <a href="https://github.com/jellyfin/jellyfin-chromecast/issues/new/choose">Issue</a> on GitHub.<br/>
```
Comment (thornbill) [inline README.md:31]:
```suggestion
* If yes: The issue probably lies somewhere else.
* If no: [Open an issue here.](https://github.com/jellyfin/jellyfin-chromecast/issues/new/choose)
```

File: README.md (hunk 1 above)
Comment (tim-vk) [reply README.md:1]: I agree `Jellyfin Chromecast Receiver (app)` would be better suited. But not sure if there is a different name that refers to this repo. Setting it to your suggestion now but i'm open to any suggestions.

File: README.md (hunk 4 above)
Comment (tim-vk) [reply README.md:16]: I've added a seperate "how do i use it". I did notice that the current contributing is not really up to date anymore. Since that refers to editing `jellyfin-web` if you want to host your own version. That's no longer possible since this mr: https://github.com/jellyfin/jellyfin-meta/issues/45

Though I'm uncertain how the workflow should look like now. But that's not an issue for this readme :) I've referred to the contributing.md.

File: README.md [hunk 8]
```diff
@@ -1,3 +1,46 @@
-# Jellyfin Chromecast
+<h1 align="center">Jellyfin Chromecast Web Receiver</h1>
+<h3 align="center">Part of the <a href="https://jellyfin.org">Jellyfin Project</a></h3>
 
-Source code for the Jellyfin Chromecast receiver application
+<p align="center">
+<img alt="Logo Banner" src="https://raw.githubusercontent.com/jellyfin/jellyfin-ux/master/branding/SVG/banner-logo-solid.svg?sanitize=true"/>
+<br/>
+<br/>
+<a href="https://github.com/jellyfin/jellyfin">
```
Comment (nielsvanvelzen) [inline README.md:8]: Should link to this repo

File: README.md [hunk 9]
```diff
@@ -1,3 +1,46 @@
-# Jellyfin Chromecast
+<h1 align="center">Jellyfin Chromecast Web Receiver</h1>
```
Comment (nielsvanvelzen) [inline README.md:1]:
```suggestion
<h1 align="center">Jellyfin Cast Web Receiver</h1>
```

Chromecast is renamed to Google Cast or just Cast these days, so let's use that for the name (even though the repository url still uses the chromecast name)

File: README.md [hunk 10]
```diff
@@ -1,3 +1,46 @@
...
+<br/>
+<a href="https://github.com/jellyfin/jellyfin">
+<img alt="GPL 2.0 License" src="https://img.shields.io/github/license/jellyfin/jellyfin-chromecast.svg"/>
+</a>
+<a href="https://github.com/jellyfin/jellyfin/releases">
+<img alt="Current Release" src="https://img.shields.io/github/release/jellyfin/jellyfin-chromecast.svg"/>
+</a>
+</p>
+
+Jellyfin Chromecast is a receiver app used when casting to a Google Cast capable device. It is used when casting from the Jellyfin Android client or Jellyfin web client.
```
Comment (nielsvanvelzen) [inline README.md:16]:
```suggestion
The Jellyfin cast web receiver is the frontend used when casting to a Google Cast capable device. It is used by default when casting from the Jellyfin Android app or Jellyfin web client.
```

File: README.md [hunk 11]
```diff
@@ -1,3 +1,46 @@
...
+
+### How do I use it?
+
+A `stable` and `unstable` version of this app are already included in `jellyfin-server`. There is no need to seperately install this project. To host your own version (for developing) see `CONTRIBUTING.md`.
+
+### What does it do?
+
+This is a `web receiver` as defined in the [Google Chromecast architecture](https://developers.google.com/cast/docs/overview).
+
+As soon as you push the "cast" button on your client this application will start on you cast-capable device & handle it from there.
```
Comment (nielsvanvelzen) [inline README.md:26]:
```suggestion
As soon as you press the "cast" button on your client this application will start on you cast-capable device and handle playback functionality. 
```

File: README.md [hunk 12]
```diff
@@ -1,3 +1,46 @@
...
+As soon as you push the "cast" button on your client this application will start on you cast-capable device & handle it from there. 
+
+### What doesn't it do?
+
+Anything related to your non-cast device (e.g. your phone, browser, other device) or anything about the inclusion of casting for a specific client (e.g. casting from the iOS app).
+
+Any issues/features related to that: check the respective repository.
+
+### Something not working right?
+First check if the issue is actually Google Cast related. So answer the question:
```
Comment (nielsvanvelzen) [inline README.md:35]:
```suggestion
### Something not working right?

First check if the issue is actually Google Cast related. So answer the question:
```

File: README.md [hunk 13]
```diff
@@ -1,3 +1,46 @@
...
+
+`"Can I reproduce the issue on any other device than a Google Chromecast?"`
```
Comment (nielsvanvelzen) [inline README.md:37]:
```suggestion
`"Can I reproduce the issue on any other device than a Google Chromecast?"`
```

My Android TV contains Chromecast functionality but it is not a Google Chromecast. Not sure how to rephrase it though.

File: README.md [hunk 14]
```diff
@@ -1,3 +1,46 @@
...
+
+If yes: The issue probably lies somewhere else. 
+If no: Open an <a href="https://github.com/jellyfin/jellyfin-chromecast/issues/new/choose">Issue</a> on GitHub.<br/>
```
Comment (nielsvanvelzen) [inline README.md:40]: Use markdown syntax for link

Comment (nielsvanvelzen) [inline README.md:40]:
```suggestion
If no: Open an <a href="https://github.com/jellyfin/jellyfin-chromecast/issues/new/choose">Issue</a> on GitHub.
```

File: README.md (hunk 13 above)
Comment (tim-vk) [reply README.md:37]: Oof... I see what you mean. Additionally: you can both cast to the android tv and run the android tv jellyfin client. (And only the first one is relevant for this repo). I've changed the wording. But open to other suggestions.

**Comment by tim-vk** [comment]
@nielsvanvelzen : Thanks for the review! I've made some changes accordingly.

Please don't forget the two questions i've put in the description:

- Current latest version is 1.0.1 however there is also a version 1.1.0 version available which seems odd. Is this a mistake? (I do think 1.0.1 is newer, but the number is off. Would assume it to be 1.1.1 or 1.2.0).
- The bit about testing: We can switch between a stable & unstable version. But no where can I actually find an actual implementation of something that does something with this setting. The jellyfin-web client seems to use it's own chromecast plugin. So is this repo even used a.t.m.? (if so: can you point me to the right direction so I can figure out stuff further from there & update the CONTRIBUTING.md accordingly?).

**Comment by nielsvanvelzen** [comment]
Thanks for the reminder about your questions, I skipped the description 😅

> Current latest version is 1.0.1 however there is also a version 1.1.0 version available which seems odd. Is this a mistake? (I do think 1.0.1 is newer, but the number is off. Would assume it to be 1.1.1 or 1.2.0).

I'm honestly not sure what happened as I didn't do anything with this repo when those releases were made. My best guess is that 1.0.1 was incorrectly labelled and should've been 1.1.1. The diff appears to contain three commits: https://github.com/jellyfin/jellyfin-chromecast/compare/v1.1.0...v1.0.1. When I look at the source of the deployed CC app I can see the added comment in embyactions.js so 1.0.1 is definitely the "latest" and deployed version.

> The bit about testing: We can switch between a stable & unstable version. But no where can I actually find an actual implementation of something that does something with this setting. The jellyfin-web client seems to use it's own chromecast plugin. So is this repo even used a.t.m.? (if so: can you point me to the right direction so I can figure out stuff further from there?).

We've made changes for 10.9 that make it possible to add your own receiver. Documentation for that is in jellyfin/jellyfin-meta#45. Note that the actual implementation might differ from the proposal.

**Review by nielsvanvelzen — APPROVED** [review]
Awesome, thanks for your work!
